ssz/codec.py

```diff
@@ -0,0 +1,37 @@
+from functools import (
+    lru_cache,
+)
+
+from eth_utils import (
+    is_bytes,
+)
+
+from ssz.exceptions import (
+    DecodingError,
+)
+from ssz.utils import (
+    infer_sedes,
+)
+
+
+@lru_cache(128)
```

Comment:
I'm not convinced that we need this optimization and would rather remove it until we're sure that it actually makes a difference.